docs/proposals/odo-config-file.md

```diff
@@ -0,0 +1,222 @@
+
+* app-name: (local) Preferred application name if none is defined
+* component-name: (local) Preferred component name if none is defined
+* odoignore: (local/global) list of files/patterns to ignore by default if no .odoignore file is provided, or when it's cerated. **This configuration can be global, as the developer might want to always ignore specific files (e.g. .git, .odo, ...) when pushing.**
+
+#### <a name="odo-file-format"></a> `odo` file format
+
+The format for both files should be human readable. Whether it's yaml, json or [toml](https://github.com/toml-lang/toml) is not the scope of this proposal to define, as it will be dictated by the engineering team implementing this feature. In any case, it should be possible to:
+
+* Visually interpret the content of a file (vi, cat)
+* Process the file in an easy way (grep, aqk, sed, jq|yq)
+* Be stored (and visualized) in version control system
+* Be used in any operating system (Linux, mac OS, windows)
+
+Nevertheless, `odo` will provide a way to manipulate this file via a CLI command, as described in [managing configuration](#managing-configuration)
+
+### <a name="managing-configuration"></a> Managing configuration
+
+Developers will need to add configuration values to the `.odo` config files. This configuration management will be managed by a new command in `odo` CLI. As configuration will be of different types, there will be needs to additional verbs (to the regular ones) for managing this configuration.
+
+#### Creating configuration
+
+There might be times that a developer would want to add/edit/remove values from the .odo configuration file. There should be a command to manage this file in the `odo` CLI. That command should be the same used to touch configuration of the global `odo` config file located in the users's home directory.
+
+Currently that command is:
+
+```bash
+odo utils config [create|update|delete|list|add|remove|get|describe] [KEY] [VALUES] [--local|--local]
+```
+
+In order to create configuration, we should call odo with the appropriate key and value or list of values to apply, as well as the scope for this configuration to be saved (local or global).
+
+```bash
+odo utils config set memory 2Gi               # Set's component memory to 2 GB local config
+odo utils config add odoignore .odo --local   # Adds a value to odoignore list local config
+odo utils config add odoignore .odo --global  # Adds a value to odoignore list global config
+```
+
+As can be seen, when a scope is ommited, the value will be applied to the local configuration store.
+
+The verbs that add configuration are:
+
+* *create*: Creates a configuration entry with the value provided. If the key exists, this command should error with the corresponding message.
+* *add*: Adds a value to an existing config entry. The type of the entry should be a list. If the key does not exist, or is not a list, this command should error with the corresponding error message.
```

Comment:
Not very sure, but can it be append(instead of add), so that it becomes more indicative of appending a value to list?

Reply:
Add is more intuitive, at least in Java that's the ver one would use to add to a list when order don't matter